<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Cryption_toolClass</class>
 <widget class="QMainWindow" name="Cryption_toolClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>922</width>
    <height>621</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cryption_tool</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>881</width>
      <height>561</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>DES</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget_4">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>811</width>
           <height>461</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_12">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_16">
            <item>
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>请输入密钥：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_6"/>
            </item>
            <item>
             <spacer name="horizontalSpacer_22">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <spacer name="horizontalSpacer_23">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_24">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_17">
            <item>
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>原文：</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_25">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label_19">
              <property name="text">
               <string>密文：</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_26">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_18">
            <item>
             <widget class="QTextEdit" name="textEdit"/>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_13">
              <item>
               <widget class="QPushButton" name="pushButton_16">
                <property name="text">
                 <string>加密</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_17">
                <property name="text">
                 <string>解密</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_18">
                <property name="text">
                 <string>清除</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTextEdit" name="textEdit_2"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>SM4</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget_3">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>20</y>
           <width>811</width>
           <height>461</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>请输入密钥：</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_4"/>
            </item>
            <item>
             <spacer name="horizontalSpacer_15">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <spacer name="horizontalSpacer_16">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_12">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>原文：</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_13">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>密文：</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_14">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_12">
            <item>
             <widget class="QTextEdit" name="textEdit_3"/>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QPushButton" name="pushButton_10">
                <property name="text">
                 <string>加密</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_11">
                <property name="text">
                 <string>解密</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_12">
                <property name="text">
                 <string>清除</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTextEdit" name="textEdit_4"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>922</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="Cryption_tool.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_16</sender>
   <signal>clicked()</signal>
   <receiver>Cryption_toolClass</receiver>
   <slot>encryption()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_18</sender>
   <signal>clicked()</signal>
   <receiver>textEdit_2</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>469</x>
     <y>486</y>
    </hint>
    <hint type="destinationlabel">
     <x>573</x>
     <y>521</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_17</sender>
   <signal>clicked()</signal>
   <receiver>Cryption_toolClass</receiver>
   <slot>decryption()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>379</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>623</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_18</sender>
   <signal>clicked()</signal>
   <receiver>textEdit</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>484</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>525</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>encryption()</slot>
  <slot>decryption()</slot>
 </slots>
</ui>
